cli/azd/internal/telemetry/resource/installed_by.go

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package resource
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+const cInstalledByFileName = ".installed-by.txt"
+
+// Returns content of `.installed-by.txt` file in the same directory as
+// the executable. If the file does not exist, returns empty string.
+func getInstalledBy() string {
+	exePath, err := os.Executable()
+
+	if err != nil {
+		return ""
+	}
+
+	resolvedPath, err := filepath.EvalSymlinks(exePath)
+	if err != nil {
+		return ""
+	}
+
+	exeDir := filepath.Dir(resolvedPath)
+	installedByFile := filepath.Join(exeDir, cInstalledByFileName)
+
+	bytes, err := os.ReadFile(installedByFile)
+	if err != nil {
+		return ""
+	}
+
+	return strings.TrimSpace(string(bytes))
```

Comment:
Can we hash the output here? I know this is technically not supposed to be modified by users, but I suppose it is a file on the filesystem can be modified, and our current telemetry guidance is to deanonymize any user data.

Reply:
I think there are compliance policies that prevent us from logging user data based on how our data store is currently set up. I think this maybe falls under that category, given that we are reading from a file owned by a user on a filesystem (it is correct that we bootstrap the creation of this file).

So I would hash the output, and keep track of the list of hashes in telemetry (we have functions that can we easily write to this). Other hashes would be listed as Unknown. It's trivial to update the list of hash mappings.